sdk/src/workflow_context.rs

```diff
@@ -370,6 +367,29 @@ impl WfContext {
     }
 }
 
+/// Helper Wrapper that can drain the channel into a Vec<SignalData> in a blocking way.  Useful
+/// for making sure channels are empty before ContinueAsNew-ing a workflow
+pub struct DrainableSignalStream(UnboundedReceiverStream<SignalData>);
+
+impl DrainableSignalStream {
+    pub fn drain_all(mut self) -> Vec<SignalData> {
+        let mut cx = std::task::Context::from_waker(futures::task::noop_waker_ref());
+        let mut signals = vec![];
+        while let Poll::Ready(Some(s)) = self.0.poll_next_unpin(&mut cx) {
```

Comment:
I think you can avoid the no-op waker if you turn the stream back into a receiver https://docs.rs/tokio-stream/latest/tokio_stream/wrappers/struct.UnboundedReceiverStream.html#method.into_inner and then just call `try_recv` in a loop

Reply:
fixed